vidarr-pluginapi/src/main/java/ca/on/oicr/gsi/vidarr/InputType.java

```diff
@@ -684,15 +704,44 @@ public static InputType dictionary(InputType key, InputType value) {
    * Create a new object type
    *
    * @param fields the names and types of the fields; duplicate names are not permitted
+   * @throws IllegalArgumentException if given a Stream with 0 elements, or a field key is empty
+   *     string
    */
   public static InputType object(Stream<Pair<String, InputType>> fields) {
-    return new ObjectInputType(fields);
+    // Sanity checking
+    List fieldsList = fields.collect(Collectors.toList());
+    if (fieldsList.size() == 0)
+      throw new IllegalArgumentException("Object InputType needs at least 1 field, got 0.");
+
+    List keys =
+        ((List)
+            fieldsList.stream()
+                .map((p) -> ((Pair<String, InputType>) p).first())
+                .collect(Collectors.toList()));
+    if (keys.contains(""))
+      throw new IllegalArgumentException(
+          "Found illegal field key \"\" while creating Object InputType.");
+
+    Set dupes =
```

Comment:
This is a costly way to get the duplicates. Try something like:

```
for (final var entry :fieldList.stream().collect(Collectors.groupingBy(Pair::first, Collectors.counting())).entrySet()) {
  if (entry.getValue() > 1) {
throw ...
}
}
```
You could also move the `""` check in the loop and avoid creating that list.

Reply:
Fixed!